### PR TITLE
Add documentation examples for `raising-bad-type`

### PR DESCRIPTION
**Pull request by julthep**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description
Add documentation examples for `raising-bad-type`

Refs #5953

### CODE REVIEW

File: pylint/checkers/exceptions.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def _is_raising(body: list) -> bool:
     "E0702": (
         "Raising %s while only classes or instances are allowed",
         "raising-bad-type",
-        "Used when something which is neither a class nor an instance "
+        "Used when something which is neither a class, an instance nor a string "
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/exceptions.py:72]: I don't understand this change. I don't think you can raise a string ?
```
>>> raise "A string"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: exceptions must derive from BaseException
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: pylint/checkers/exceptions.py (hunk 1 above)
Comment (julthep) [reply pylint/checkers/exceptions.py:72]: This addition comes from [pylint-errors E0702](https://vald-phoenix.github.io/pylint-errors/plerr/errors/exceptions/E0702). Should it be reverted?

Comment (Pierre-Sassoulas) [reply pylint/checkers/exceptions.py:72]: Yeap I think it's wrong.
```suggestion
        "Used when something which is neither a class nor an instance "
```

File: doc/data/messages/r/raising-bad-type/bad.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+def div(x, y):
+    try:
+        return x / y
+    except ZeroDivisionError:
+        raise None  # [raising-bad-type]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/r/raising-bad-type/bad.py:5]:
```suggestion
class FasterThanTheSpeedOfLightError(ZeroDivisionError):
    def __init__(self):
        super().__init__("You can't go faster than the speed of light !")


def calculate_speed(distance: float, time: float) -> float:
    try:
        return distance / time
    except ZeroDivisionError as e:
        raise None  # [raising-bad-type]
```

File: doc/data/messages/r/raising-bad-type/good.py [hunk 3]
```diff
@@ -1 +1,5 @@
-# This is a placeholder for correct code for this message.
+def div(x, y):
+    try:
+        return x / y
+    except ZeroDivisionError as e:
+        raise ValueError from e
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/r/raising-bad-type/good.py:5]:
```suggestion
class FasterThanTheSpeedOfLightError(ZeroDivisionError):
    def __init__(self):
        super().__init__("You can't go faster than the speed of light !")


def calculate_speed(distance: float, time: float) -> float:
    try:
        return distance / time
    except ZeroDivisionError as e:
        raise FasterThanTheSpeedOfLightError() from e
```

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**